libs/usb-drive/src/mock_usb_drive.ts

```diff
@@ -0,0 +1,88 @@
+import { MockFunction, mockFunction } from '@votingworks/test-utils';
+import { execSync } from 'child_process';
+import { join } from 'path';
+import { mkdirSync, writeFileSync } from 'fs';
+import tmp from 'tmp';
+import { Buffer } from 'buffer';
+import { UsbDrive } from './usb_drive';
+
+type MockFileTree = MockFile | MockDirectory;
+type MockFile = Buffer;
+interface MockDirectory {
+  [name: string]: MockFileTree;
+}
+
+function writeMockFileTree(destinationPath: string, tree: MockFileTree): void {
+  if (Buffer.isBuffer(tree)) {
+    writeFileSync(destinationPath, tree);
+  } else {
+    mkdirSync(destinationPath, { recursive: true });
+    for (const [name, child] of Object.entries(tree)) {
+      // Sleep 1ms to ensure that each file is created with a distinct timestamp
+      execSync('sleep 0.01');
+      writeMockFileTree(join(destinationPath, name), child);
+    }
+  }
+}
+
+type MockedUsbDrive = {
+  [Method in keyof UsbDrive]: MockFunction<UsbDrive[Method]>;
+};
+
+/**
+ * A mock of the UsbDrive interface. See createMockUsbDrive for details.
+ */
+export interface MockUsbDrive {
+  usbDrive: MockedUsbDrive;
+  assertComplete(): void;
+  insertUsbDrive(contents: MockFileTree): void;
+  removeUsbDrive(): void;
+}
+
+/**
+ * Creates a mock of the UsbDrive interface. Each method is mocked with a
+ * mockFunction (see @votingworks/test-utils).
+ *
+ * Also has a insert()/remove() interface to create a fake USB drive backed by a
+ * filesystem directory. If using this interface, the mock functions will
+ * automatically be updated to return the correct status.
+ *
+ * Warning: if you use both interfaces, they may get out of sync.
+ */
+export function createMockUsbDrive(): MockUsbDrive {
+  let mockUsbTmpDir: tmp.DirResult | undefined;
+
+  const usbDrive: MockedUsbDrive = {
+    status: mockFunction('status'),
+    eject: mockFunction('eject'),
+  };
+
+  return {
+    usbDrive,
+
+    assertComplete() {
+      for (const method of Object.values(usbDrive)) {
+        method.assertComplete();
+      }
+    },
+
+    insertUsbDrive(contents: MockFileTree) {
+      mockUsbTmpDir?.removeCallback();
+      mockUsbTmpDir = tmp.dirSync({ unsafeCleanup: true });
```

Comment:
It's worth checking if `unsafeCleanup` actually works. I think I checked once and found it did not, because of the way `jest` exits (or fails to exit).

Reply:
Good callout. Do you have ideas for how to make this cleanup well otherwise though? Manual cleanup is tough to enforce. I'm kind of wondering if letting it leak is really that bad, since the worst case is just using up a lot of space on a dev machine.

Reply:
It's not necessary, for sure. our new CVR files are bigger and I've hit low disk space once before, and had to manually cleanup. if something like this were easy, it would be a nice to have, but i'd have to think about how this is being run within tests:
https://github.com/votingworks/vxsuite/blob/6fe6a1e3efb0b3cac757f692596e62f1bc91111b/libs/fixtures/src/data/electionMinimalExhaustiveSample/cvr-files/standard.ts#L11-L17

Reply:
Added cleanup here: 1178426b2. Thanks for the suggestion!